validation/src/main/scala/hmda/validation/rules/lar/validity/V290.scala

```diff
@@ -0,0 +1,38 @@
+package hmda.validation.rules.lar.validity
+
+import hmda.model.census.CBSATractLookup
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.dsl.{ Failure, Result }
+import hmda.validation.rules.EditCheck
+
+object V290 extends EditCheck[LoanApplicationRegister] {
+
+  val cbsaTracts = CBSATractLookup.values
+
+  val validCombinations = cbsaTracts.map { cbsa =>
+    (cbsa.state, cbsa.county, cbsa.geoidMsa)
+  }
+
+  override def name: String = "V290"
+
+  override def apply(input: LoanApplicationRegister): Result = {
+    val msa = msaCode(input.geography.msa)
+    val state = input.geography.state
+    val county = input.geography.county
+
+    val combination = (state, county, msa)
+
+    when(msa not equalTo("NA")) {
+      combination is containedIn(validCombinations)
+    }
+  }
+
+  private def msaCode(code: String): String = {
+    val md = cbsaTracts.filter(m => m.metDivFp == code)
```

Comment:
I'm a little concerned about the performance of this search (and the one above). 70k is not a huge number, but it's not tiny either. It might be premature to optimize it now, but we might want to keep a list of things to investigate first if performance becomes a problem when we start measuring it.